src/test/java/github/grace000/erversay/MockConnection.java

```diff
@@ -0,0 +1,40 @@
+package github.grace000.erversay;
+
+public class MockConnection implements ConnectionWrapper {
+    private boolean listenForConnectionCalled = false;
+    private boolean acceptConnectionCalled = false;
+    private boolean getIOStreamsCalled = false;
+    private boolean writeResponseCalled = false;
+
+    public void listenForConnection(int port) {
+        listenForConnectionCalled = true;
+    }
+
+    public void serveClient(String response) {
+        acceptConnectionCalled = true;
+        getIOStreamsCalled = true;
+        writeResponseCalled = true;
+    }
```

Comment:
I don't love how you have three things set to true here. After all, if you change it in your code but not your tests, the tests won't reflect that.

Reply:
These methods are now all broken up into single methods. I can test that each was called.